src/utils/utils.cpp

```diff
@@ -212,6 +216,55 @@ void mp::utils::wait_for_cloud_init(mp::VirtualMachine* virtual_machine, std::ch
     mp::utils::try_action_for(on_timeout, timeout, action);
 }
 
+void mp::utils::install_sshfs_for(const std::string& name, mp::SSHSession& session)
+{
+    mpl::log(mpl::Level::info, category, fmt::format("Installing the multipass-sshfs snap in \'{}\'", name));
+
+    // Check if snap support is installed in the instance
+    auto which_proc = session.exec("which snap");
+    if (which_proc.exit_code() != 0)
+    {
+        mpl::log(mpl::Level::warning, category, fmt::format("Snap support is not installed in \'{}\'", name));
+        throw std::runtime_error(
+            fmt::format("Snap support needs to be installed in \'{}\' in order to support mounts.\n"
+                        "Please see https://docs.snapcraft.io/installing-snapd for information on\n"
+                        "how to install snap support for your instance's distribution.\n\n"
+                        "If your distribution's instructions specify enabling classic snap support,\n"
+                        "please do that as well.\n\n"
+                        "Alternatively, install `sshfs` manually inside the instance.",
+                        name));
+    }
+
+    // Check if /snap exists for "classic" snap support
+    auto test_file_proc = session.exec("[ -e /snap ]");
+    if (test_file_proc.exit_code() != 0)
+    {
+        mpl::log(mpl::Level::warning, category, fmt::format("Classic snap support symlink is needed in \'{}\'", name));
+        throw std::runtime_error(
+            fmt::format("Classic snap support is not enabled for \'{}\'!\n\n"
+                        "Please see https://docs.snapcraft.io/installing-snapd for information on\n"
+                        "how to enable classic snap support for your instance's distribution.",
+                        name));
+    }
+
+    try
+    {
+        auto proc = session.exec("sudo snap install multipass-sshfs");
+        if (proc.exit_code(std::chrono::minutes(5)) != 0)
+        {
+            auto error_msg = proc.read_std_error();
+            mpl::log(mpl::Level::warning, category,
+                     fmt::format("Failed to install \'multipass-sshfs\', error message: \'{}\'",
+                                 mp::utils::trim_end(error_msg)));
```

Comment:
Hmm this didn't bubble up when [the Fedora](https://forum.snapcraft.io/t/installing-snap-on-fedora/6755/17) mount failed…

I suppose that's due to our log forwarding thing not working (#962, but maybe more…).

Reply:
Right, log messages...it's log level `warning` for this too, so I think that would require at least 1 `-v` to be specified for it to show in the client if log forwarding worked.